File: wagerproof-ios-native/Wagerproof/Features/LearnMore/Components/Slide5_Outliers.swift
import SwiftUI
import WagerproofDesign

/// Mirrors `wagerproof-mobile/components/learn-wagerproof/slides/Slide5_Outliers.tsx`.
///
/// Two alert cards (one VALUE, one FADE) followed by a small legend explaining
/// what each alert type means. Each card carries a pro-lock badge in the
/// top-right corner per the RN design.
struct Slide5_Outliers: View {

    private struct Alert: Identifiable {
        let id = UUID()
        let kind: Kind
        let sport: String
        let matchup: String
        let description: String
        let confidence: Int
        let suggestedBet: String?
        enum Kind { case value, fade }
    }

    private let alerts: [Alert] = [
        Alert(
            kind: .value, sport: "NFL", matchup: "Patriots @ Dolphins",
            description: "Polymarket shows 67% on Patriots +3.5",
            confidence: 67, suggestedBet: nil
        ),
        Alert(
            kind: .fade, sport: "NFL", matchup: "Bills @ Jets",
            description: "Model predicts Bills at 82% - Historical fade opportunity",
            confidence: 82, suggestedBet: "Jets +7"
        ),
    ]

    var body: some View {
        VStack(spacing: Spacing.lg) {
            VStack(spacing: Spacing.md) {
                ForEach(alerts) { alert in
                    alertCard(alert)
                }
            }

            VStack(alignment: .leading, spacing: Spacing.sm) {
                legendRow(color: Color(hex: 0x22C55E), text: "Value alerts: Market disagrees with Vegas")
                legendRow(color: Color(hex: 0xF59E0B), text: "Fade alerts: High confidence = fade opportunity")
            }
        }
    }

    private func alertCard(_ a: Alert) -> some View {
        let color: Color = a.kind == .value ? Color(hex: 0x22C55E) : Color(hex: 0xF59E0B)
        return VStack(alignment: .leading, spacing: Spacing.sm) {
            HStack(spacing: Spacing.sm) {
                Text(a.sport)
                    .font(.system(size: 10, weight: .bold))
                    .foregroundStyle(Color.appTextSecondary)
                    .padding(.horizontal, 8)
                    .padding(.vertical, 3)
                    .background(Color.appSurfaceMuted)
                    .clipShape(RoundedRectangle(cornerRadius: 6, style: .continuous))

                HStack(spacing: 4) {
                    Image(systemName: a.kind == .value ? "chart.line.uptrend.xyaxis" : "bolt.fill")
                        .font(.system(size: 10, weight: .semibold))
                        .foregroundStyle(color)
                    Text(a.kind == .value ? "VALUE" : "FADE")
                        .font(.system(size: 10, weight: .bold))
                        .foregroundStyle(color)
                }
                .padding(.horizontal, 8)
                .padding(.vertical, 3)
                .background(color.opacity(0.2))
                .clipShape(RoundedRectangle(cornerRadius: 6, style: .continuous))

                Text("\(a.confidence)%")
                    .font(.system(size: 10, weight: .bold))
                    .foregroundStyle(color)
                    .padding(.horizontal, 8)
                    .padding(.vertical, 3)
                    .background(color.opacity(0.2))
                    .clipShape(RoundedRectangle(cornerRadius: 6, style: .continuous))

                Spacer()
            }

            Text(a.matchup)
                .font(.system(size: 14, weight: .semibold))
                .foregroundStyle(Color.appTextPrimary)

            Text(a.description)
                .font(.system(size: 12))
                .lineSpacing(2)
                .foregroundStyle(Color.appTextSecondary)
                .fixedSize(horizontal: false, vertical: true)

            if let suggested = a.suggestedBet {
                HStack(spacing: 8) {
                    Text("Suggested:")
                        .font(.system(size: 11))
                        .foregroundStyle(Color.appTextSecondary)
                    Text(suggested)
                        .font(.system(size: 13, weight: .bold))
                        .foregroundStyle(color)
                }
                .padding(.horizontal, 10)
                .padding(.vertical, 8)
                .background(color.opacity(0.1))
                .clipShape(RoundedRectangle(cornerRadius: 8, style: .continuous))
                .overlay(
                    RoundedRectangle(cornerRadius: 8, style: .continuous)
                        .strokeBorder(color.opacity(0.3), lineWidth: 1)
                )
            }
        }
        .padding(Spacing.md)
        .background(Color.appSurfaceElevated)
        .clipShape(RoundedRectangle(cornerRadius: 14, style: .continuous))
        .overlay(
            RoundedRectangle(cornerRadius: 14, style: .continuous)
                .strokeBorder(color.opacity(0.25), lineWidth: 1)
        )
        .overlay(alignment: .top) {
            Rectangle()
                .fill(color)
                .frame(height: 3)
                .clipShape(
                    UnevenRoundedRectangle(
                        topLeadingRadius: 14, bottomLeadingRadius: 0,
                        bottomTrailingRadius: 0, topTrailingRadius: 14
                    )
                )
        }
        .overlay(alignment: .topTrailing) {
            HStack(spacing: 4) {
                Image(systemName: "lock.fill")
                    .font(.system(size: 9))
                    .foregroundStyle(Color.appAccentAmber)
                Text("Pro Feature")
                    .font(.system(size: 9, weight: .semibold))
                    .foregroundStyle(Color.appAccentAmber)
            }
            .padding(.horizontal, 8)
            .padding(.vertical, 4)
            .background(Color.appSurfaceMuted)
            .clipShape(RoundedRectangle(cornerRadius: 8, style: .continuous))
            .padding(10)
        }
        .shadow(color: .black.opacity(0.1), radius: 3, x: 0, y: 1)
    }

    private func legendRow(color: Color, text: String) -> some View {
        HStack(spacing: 8) {
            Circle().fill(color).frame(width: 12, height: 12)
            Text(text)
                .font(.system(size: 12))
                .foregroundStyle(Color.appTextSecondary)
            Spacer()
        }
    }
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/learn/slides/Slide5Outliers.kt
package com.wagerproof.app.features.learn.slides

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.wagerproof.core.design.icons.AppIcon
import com.wagerproof.core.design.tokens.AppColors
import com.wagerproof.core.design.tokens.Spacing

/**
 * Port of iOS `Slide5_Outliers.swift`. Two alert cards (VALUE + FADE) with
 * pro-lock badges + a green/amber legend. Hardcoded marketing content.
 */
private enum class AlertKind { VALUE, FADE }

private data class OutlierAlert(
    val kind: AlertKind,
    val sport: String,
    val matchup: String,
    val description: String,
    val confidence: Int,
    val suggestedBet: String?,
)

private val alerts = listOf(
    OutlierAlert(AlertKind.VALUE, "NFL", "Patriots @ Dolphins", "Polymarket shows 67% on Patriots +3.5", 67, null),
    OutlierAlert(AlertKind.FADE, "NFL", "Bills @ Jets", "Model predicts Bills at 82% - Historical fade opportunity", 82, "Jets +7"),
)

@Composable
fun Slide5Outliers(modifier: Modifier = Modifier) {
    Column(modifier, verticalArrangement = Arrangement.spacedBy(Spacing.lg)) {
        Column(verticalArrangement = Arrangement.spacedBy(Spacing.md)) {
            alerts.forEach { alertCard(it) }
        }
        Column(verticalArrangement = Arrangement.spacedBy(Spacing.sm)) {
            legendRow(Color(0xFF22C55E), "Value alerts: Market disagrees with Vegas")
            legendRow(Color(0xFFF59E0B), "Fade alerts: High confidence = fade opportunity")
        }
    }
}

@Composable
private fun alertCard(a: OutlierAlert) {
    val color = if (a.kind == AlertKind.VALUE) Color(0xFF22C55E) else Color(0xFFF59E0B)
    Box(
        Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(14.dp))
            .background(AppColors.appSurfaceElevated)
            .border(1.dp, color.copy(alpha = 0.25f), RoundedCornerShape(14.dp)),
    ) {
        // Top accent bar.
        Box(Modifier.fillMaxWidth().height(3.dp).background(color))
        Column(
            Modifier.padding(Spacing.md),
            verticalArrangement = Arrangement.spacedBy(Spacing.sm),
        ) {
            Row(
                horizontalArrangement = Arrangement.spacedBy(Spacing.sm),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                Text(
                    a.sport,
                    fontSize = 10.sp,
                    fontWeight = FontWeight.Bold,
                    color = AppColors.appTextSecondary,
                    modifier = Modifier
                        .clip(RoundedCornerShape(6.dp))
                        .background(AppColors.appSurfaceMuted)
                        .padding(horizontal = 8.dp, vertical = 3.dp),
                )
                Row(
                    Modifier
                        .clip(RoundedCornerShape(6.dp))
                        .background(color.copy(alpha = 0.2f))
                        .padding(horizontal = 8.dp, vertical = 3.dp),
                    horizontalArrangement = Arrangement.spacedBy(4.dp),
                    verticalAlignment = Alignment.CenterVertically,
                ) {
                    Icon(
                        imageVector = AppIcon.fromSystemName(
                            if (a.kind == AlertKind.VALUE) "chart.line.uptrend.xyaxis" else "bolt.fill",
                        )!!.imageVector,
                        contentDescription = null,
                        tint = color,
                        modifier = Modifier.size(10.dp),
                    )
                    Text(if (a.kind == AlertKind.VALUE) "VALUE" else "FADE", fontSize = 10.sp, fontWeight = FontWeight.Bold, color = color)
                }
                Text(
                    "${a.confidence}%",
                    fontSize = 10.sp,
                    fontWeight = FontWeight.Bold,
                    color = color,
                    modifier = Modifier
                        .clip(RoundedCornerShape(6.dp))
                        .background(color.copy(alpha = 0.2f))
                        .padding(horizontal = 8.dp, vertical = 3.dp),
                )
                Spacer(Modifier.weight(1f))
            }

            Text(a.matchup, fontSize = 14.sp, fontWeight = FontWeight.SemiBold, color = AppColors.appTextPrimary)
            Text(a.description, fontSize = 12.sp, color = AppColors.appTextSecondary, lineHeight = 16.sp)

            a.suggestedBet?.let { suggested ->
                Row(
                    Modifier
                        .clip(RoundedCornerShape(8.dp))
                        .background(color.copy(alpha = 0.1f))
                        .border(1.dp, color.copy(alpha = 0.3f), RoundedCornerShape(8.dp))
                        .padding(horizontal = 10.dp, vertical = 8.dp),
                    horizontalArrangement = Arrangement.spacedBy(8.dp),
                    verticalAlignment = Alignment.CenterVertically,
                ) {
                    Text("Suggested:", fontSize = 11.sp, color = AppColors.appTextSecondary)
                    Text(suggested, fontSize = 13.sp, fontWeight = FontWeight.Bold, color = color)
                }
            }
        }
        // Pro-lock badge, top-right.
        Row(
            Modifier
                .align(Alignment.TopEnd)
                .padding(10.dp)
                .clip(RoundedCornerShape(8.dp))
                .background(AppColors.appSurfaceMuted)
                .padding(horizontal = 8.dp, vertical = 4.dp),
            horizontalArrangement = Arrangement.spacedBy(4.dp),
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Icon(
                imageVector = AppIcon.fromSystemName("lock.fill")!!.imageVector,
                contentDescription = null,
                tint = AppColors.appAccentAmber,
                modifier = Modifier.size(9.dp),
            )
            Text("Pro Feature", fontSize = 9.sp, fontWeight = FontWeight.SemiBold, color = AppColors.appAccentAmber)
        }
    }
}

@Composable
private fun legendRow(color: Color, text: String) {
    Row(
        horizontalArrangement = Arrangement.spacedBy(8.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Box(Modifier.size(12.dp).clip(CircleShape).background(color))
        Text(text, fontSize = 12.sp, color = AppColors.appTextSecondary)
        Spacer(Modifier.weight(1f))
    }
}
